docs/cloudevents/cloudevents-spans.md

```diff
@@ -35,168 +31,38 @@ consult the
-|    |     +---------------------------------------+    |
-|    |     +---------------------------------------+    |
-|    |     | Send (auto-instr)                     |    |
-|    |     +---------------------------------------+    |
-|    |        +------------------------------------+    |
-|    |        | Intermediary: Received (auto-instr)|    |
-|    |        +------------------------------------+    |
-|    |        +------------------------------------+    |
-|    |        | Intermediary: Send (auto-instr)    |    |
-|    |Link    +------------------------------------+    |
-|    |                                                  |
-|    |                                                  |
-|    |                                                  |
-|    |     Trace 2                                      |
-|    |                                                  |
-|    |     +---------------------------------------+    |
-|    |     | Consumer: Receive (auto-instr)        |    |
-|    |     +---------------------------------------+    |
-|    |       +-------------------------------------+    |
-|    +------ | Consumer: Process                   |    |
-|            +-------------------------------------+    |
-|                                                       |
-+-------------------------------------------------------+
-```
-
 ## Conventions
 
-To achieve the trace above, it is necessary to capture the context of
-the event creation so that when the CloudEvent reaches its destination(s), this
-context can be continued. Each CloudEvent acts then as the medium of this
-context propagation.
-
-This document relies on the CloudEvents specification, which defines this
-context propagation mechanism via the
-[CloudEvents Distributed Tracing Extension](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/extensions/distributed-tracing.md).
-Once the trace context is set on the event
-via the Distributed Tracing Extension, it MUST not be modified.
-
-The remainder of this section describes the semantic conventions for Spans
-required to achieve the proposed trace.
-
-### Spans
-
-#### Creation
-
-Instrumentation SHOULD create a new span and populate the
-[CloudEvents Distributed Tracing Extension](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/extensions/distributed-tracing.md)
-on the event. This applies when:
-
-- A CloudEvent is created by the instrumented library.
-  It may be impossible or impractical to create the Span during event
-  creation (e.g. inside the constructor or in a factory method),
-  so instrumentation MAY create the Span later, when passing the event to the transport layer.
-- A CloudEvent is created outside of the instrumented library
-  (e.g. directly constructed by the application owner, without calling a constructor or factory method),
-  and passed without the Distributed Tracing Extension populated.
+CloudEvents can be sent via various transports, such as HTTP or via messaging
```

Comment:
transport context != message context, so the protocol used to carry data is not relevant for messaging or CloudEvents

Reply:
Not sure I got it? What I tried to say here and that one can sent CloudEvents via different transport/protocols, so http, messaging, etc and that each of those defines its own way of context propagation. Maybe I should say MAY define its own mechanism instead, as you pointed out some don't define anything. The way it is written now implies that all do which is false. Is that what you meant or?